src/go/configgenerator/listener_generator.go

```diff
@@ -365,8 +358,16 @@ func makePathMatcherFilter(serviceInfo *sc.ServiceInfo) *hcmpb.HttpFilter {
 	return pathMatcherFilter
 }
 
-func hasPathParameter(httpPattern string) bool {
-	return strings.ContainsRune(httpPattern, '{')
+func needPathRewrite(serviceInfo *sc.ServiceInfo) bool {
+	for _, method := range serviceInfo.Methods {
+		// Determined by the method only. Hence just pass in the first httpRule.
```

Comment:
Sorry, it's not clear to me why we only use the first method. 

Here's an example where i think this doesn't work: The first http rule could APPEND with empty path, so path rewrite is not needed. But the second rule may be something with a path prefix, requiring path rewrite.

If this example is correct, try to add a unit test for it.

Reply:
Hmm.  I feel it is pre-mature optimization.  ConfigMgr doens't need to be fast.  So I change it to check every httpRule